provider/keycloak_oidc_identity_provider.go

```diff
@@ -0,0 +1,111 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func resourceKeycloakOidcIdentityProvider() *schema.Resource {
+	oidcSchema := map[string]*schema.Schema{
+		"backchannel_supported": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Does the external IDP support backchannel logout?",
+		},
+		"validate_signature": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Enable/disable signature validation of SAML responses.",
+		},
+		"authorization_url": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "OIDC authorization URL.",
+		},
+		"client_id": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Client ID.",
+		},
+		"client_secret": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Sensitive:   true,
+			Description: "Client Secret.",
+		},
+		"disable_user_info": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Disable User Info.",
+		},
+		"hide_on_login_page": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Hide On Login Page.",
+		},
+		"token_url": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Token URL.",
+		},
+		"login_hint": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Login Hint.",
+		},
+		"ui_locales": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Pass current locale to identity provider",
+		},
+	}
+	oidcResource := resourceKeycloakIdentityProvider()
+	oidcResource.Schema = mergeSchemas(oidcResource.Schema, oidcSchema)
+	oidcResource.Create = resourceKeycloakIdentityProviderCreate(getOidcIdentityProviderFromData, setOidcIdentityProviderData)
+	oidcResource.Read = resourceKeycloakIdentityProviderRead(setOidcIdentityProviderData)
+	oidcResource.Update = resourceKeycloakIdentityProviderUpdate(getOidcIdentityProviderFromData, setOidcIdentityProviderData)
+	return oidcResource
+}
+
+func getOidcIdentityProviderFromData(data *schema.ResourceData) (*keycloak.IdentityProvider, error) {
+	rec, _ := getIdentityProviderFromData(data)
+	rec.ProviderId = "oidc"
+	rec.Config = &keycloak.IdentityProviderConfig{
+		BackchannelSupported: keycloak.KeycloakBoolQuoted(data.Get("backchannel_supported").(bool)),
+		UseJwksUrl:           keycloak.KeycloakBoolQuoted(true),
```

Comment:
I left this comment originally and I don't think it was addressed - I don't think this attribute is useful without the ability to customize the jwks url (correct me if I'm wrong here).  If this is the case, I think this attribute should either be removed or updated to allow users to customize this url

Reply:
Removed